### ISSUE TITLE
'Unable to load one or more of the requested types.' for IApiVersionDescriptionProvider.ApiVersionDescriptions

### ISSUE THREAD

**Issue by Fagwin** [microsoft/aspnet-api-versioning#537]
After updating .net core 3.0 preview 8, I get this error in my method Startup.Configure: 'Unable to load one or more of the requested types.' for IApiVersionDescriptionProvider.ApiVersionDescriptions

This is my code:
 public void Configure(IApplicationBuilder app, IHostingEnvironment env, IApiVersionDescriptionProvider provider)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }
            app.UseCors(MyAllowSpecificOrigins);
            app.UseStaticFiles();
            // Enable the Swagger UI middleware and the Swagger generator
            app.UseMvc();
            app.UseSwagger();
            app.UseSwaggerUI(
                options =>
                {
                    foreach (var description in provider.ApiVersionDescriptions)
                    {
                        options.SwaggerEndpoint(
                            $"/swagger/{description.GroupName}/swagger.json",
                            description.GroupName.ToUpperInvariant());
                        options.RoutePrefix = String.Empty;
                    }
                });
        } 

This is my complete error:
Exception has occurred: CLR/System.Reflection.ReflectionTypeLoadException
Une exception de type 'System.Reflection.ReflectionTypeLoadException' s'est produite dans System.Private.CoreLib.dll mais n'a pas été gérée dans le code utilisateur: 'Unable to load one or more of the requested types.'
   à System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   à System.Reflection.RuntimeAssembly.get_DefinedTypes()
   à Microsoft.AspNetCore.Mvc.ApplicationParts.AssemblyPart.get_Types()
   à Microsoft.AspNetCore.Mvc.Controllers.ControllerFeatureProvider.PopulateFeature(IEnumerable`1 parts, ControllerFeature feature)
   à Microsoft.AspNetCore.Mvc.ApplicationParts.ApplicationPartManager.PopulateFeature[TFeature](TFeature feature)
   à Microsoft.AspNetCore.Mvc.ApplicationModels.ControllerActionDescriptorProvider.GetControllerTypes()
   à Microsoft.AspNetCore.Mvc.ApplicationModels.ControllerActionDescriptorProvider.GetDescriptors()
   à Microsoft.AspNetCore.Mvc.ApplicationModels.ControllerActionDescriptorProvider.OnProvidersExecuting(ActionDescriptorProviderContext context)
   à Microsoft.AspNetCore.Mvc.Infrastructure.DefaultActionDescriptorCollectionProvider.UpdateCollection()
   à Microsoft.AspNetCore.Mvc.Infrastructure.DefaultActionDescriptorCollectionProvider.Initialize()
   à Microsoft.AspNetCore.Mvc.Infrastructure.DefaultActionDescriptorCollectionProvider.get_ActionDescriptors()
   à Microsoft.AspNetCore.Mvc.ApiExplorer.DefaultApiVersionDescriptionProvider.EnumerateApiVersions(IActionDescriptorCollectionProvider actionDescriptorCollectionProvider)
   à Microsoft.AspNetCore.Mvc.ApiExplorer.DefaultApiVersionDescriptionProvider.LazyApiVersionDescriptions.EnumerateApiVersions()
   à System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   à System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   à System.Lazy`1.CreateValue()
   à System.Lazy`1.get_Value()
   à Microsoft.AspNetCore.Mvc.ApiExplorer.DefaultApiVersionDescriptionProvider.get_ApiVersionDescriptions()
   à SAIL.cloud.Api.Startup.<>c__DisplayClass7_0.<Configure>b__0(SwaggerUIOptions options) dans C:\Users\lfagnon\OneDrive\Source\Repos\SAIL.cloud.Api\SAIL.cloud.Api\Startup.cs :ligne 205
   à Microsoft.AspNetCore.Builder.SwaggerUIBuilderExtensions.UseSwaggerUI(IApplicationBuilder app, Action`1 setupAction)
   à SAIL.cloud.Api.Startup.Configure(IApplicationBuilder app, IHostingEnvironment env, IApiVersionDescriptionProvider provider) dans \SAIL.cloud.Api\Startup.cs :ligne 202
   à System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   à System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
 à Microsoft.AspNetCore.Hosting.ConfigureBuilder.Invoke(Object instance, IApplicationBuilder builder)
   à Microsoft.AspNetCore.Hosting.ConfigureBuilder.<>c__DisplayClass4_0.<Build>b__0(IApplicationBuilder builder)
   à Microsoft.AspNetCore.Hosting.GenericWebHostBuilder.<>c__DisplayClass13_0.<UseStartup>b__2(IApplicationBuilder app)
   à Microsoft.Extensions.DependencyInjection.AutoRegisterMiddleware.<>c__DisplayClass4_0.<Configure>b__0(IApplicationBuilder app)
   à Microsoft.AspNetCore.Mvc.Filters.MiddlewareFilterBuilderStartupFilter.<>c__DisplayClass0_0.<Configure>g__MiddlewareFilterBuilder|0(IApplicationBuilder builder)
   à Microsoft.AspNetCore.HostFilteringStartupFilter.<>c__DisplayClass0_0.<Configure>b__0(IApplicationBuilder app)
   à Microsoft.AspNetCore.Hosting.GenericWebHostService.<StartAsync>d__31.MoveNext()

**Comment by commonsensesoftware** [comment]
Do you have a simple repro you can share? Troubleshooting previews is difficult. It would also be useful to know which runtime and tool versions you're using. For example `dotnet --version` and `dotnet --info`.

You stated:

>_"After updating .net core 3.0 preview 8..."_

Does that mean you were using the package published manually in the API versioning 4.0 preview release? Have you updated to the official, preview package published on NuGet? Any other source or method of referencing the libraries in Preview 8+ is unsupported.

Thanks

**Comment by Fagwin** [comment]
> Do you have a simple repro you can share? Troubleshooting previews is difficult. It would also be useful to know which runtime and tool versions you're using. For example `dotnet --version` and `dotnet --info`.

No it's a big repo, it's a company application.
This is my runtim and tools versions:
```
dotnet --info
SDK .NET Core (reflétant tous les global.json) :
 Version:   3.0.100-preview8-013656
 Commit:    8bf06ffc8d

Environnement d'exécution :
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.0.100-preview8-013656\

Host (useful for support):
  Version: 3.0.0-preview8-28405-07
  Commit:  d01b2fb7bc

.NET Core SDKs installed:
  2.2.204 [C:\Program Files\dotnet\sdk]
  3.0.100-preview7-012821 [C:\Program Files\dotnet\sdk]
  3.0.100-preview8-013656 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0-preview7.19365.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0-preview8.19405.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0-preview8-28405-07 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.0-preview8-28405-07 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

> Have you updated to the official, preview package published on NuGet?
Yes, this is my package references:
`<PackageReference Include="Microsoft.AspNetCore.Mvc.Versioning" Version="4.0.0-preview8.19405.7" />`

**Comment by commonsensesoftware** [comment]
Interesting. Your tooling is newer than even what I've used. Things work for me both locally and on the build server, which only uses the official Preview 8 build.

Is there any chance your build output is stale? To be sure, I would recommend completely deleting all **bin** and **obj** folders. While you can run `dotnet clean`, I've historically found that _clean_ operations do not always clean out everything. From a clean slate, `dotnet restore` and `dotnet build` _should_ work.

If that still doesn't work, have you looked at or tried running the [4.0 Swagger Sample](../blob/release/4.0/samples/aspnetcore/SwaggerSample)? That should demonstrate all of the constituent pieces hooked up together and running successfully. It _might_ help identify what's different in your solution.

**Comment by commonsensesoftware** [comment]
Were you ever able to solve this? Thanks.

**Comment by commonsensesoftware** [comment]
Not sure if you've fixed this, but the official 4.0 release has been published. It would appear you resolved the issue or chose another alternative. Feel free to come back or reopen the issue if things persist. Thanks.
